autoscheduler/scraper/management/commands/scrape_grades.py

```diff
@@ -132,30 +132,47 @@ def scrape_pdf(grade_dists: List[pdf_parser.GradeData], term: str) -> List[Grade
     # The grade models to return to be saved later
     scraped_grades = []
 
-    for grade_data in grade_dists:
-        dept = grade_data.dept
-
-        try:
-            # Throws a ObjectDoesNotExist error upon failure
-            section = Section.objects.get(
-                subject=dept,
-                course_num=grade_data.course_num,
-                section_num=grade_data.section_num,
-                term_code=int(term)
-            )
+    # Collect attributes into a list so we can bulk query for the Sections
+    subjects, course_nums, section_nums = zip(*((data.dept, data.course_num,
+                                                 data.section_num)
+                                                for data in grade_dists))

```

Comment:
We have to check if `grade_dists` is `None` first, otherwise it will try to extract an empty iterable into a 3-tuple and throw an error (try running the command with `-c UT`)

Reply:
Ohhh ok I was wondering what was causing that error in #215